conftest.py

```diff
@@ -210,6 +210,8 @@ def mesh_device(request, silicon_arch_name, silicon_arch_wormhole_b0, device_par
     mesh_device = ttnn.open_mesh_device(mesh_shape, dispatch_core_type=get_dispatch_core_type(), **device_params)
 
     logger.debug(f"multidevice with {mesh_device.get_num_devices()} devices is created")
+    ttnn.SetDefaultDevice(mesh_device.get_device(0))
```

Comment:
Other functions around this one do not set the device as the default one. This change makes it inconsistent.

Reply:
An alternate approach: https://github.com/tenstorrent/tt-metal/commit/79031d8dd99bcb67116c4b89ba1d5f5cb96f1d25

Reply:
This checks the default device first, but if that's not set, checks the `ARCH_NAME` env flag.